File: onboarding/src/indexing-relevance/scala/src/main/scala/com/algolia/Indexing.scala
package com.algolia

import algolia.AlgoliaClient
import algolia.AlgoliaDsl._

import scala.io.Source
import org.json4s._
import org.json4s.native.JsonMethods._

object Indexing {

  def main(args: Array[String]): Unit = {
    val json = Source.fromURL("https://alg.li/doc-ecommerce.json").bufferedReader()
    val products = parse(json).extract[Iterable[Product]]

    val client = new AlgoliaClient("YourApplicationID", "YourAPIKey")
    for {
      result <- client.execute(index into "demo_ecommerce" objects products)
    } yield result
  }

}

File: onboarding/src/indexing-relevance/java/src/main/java/com/algolia/Indexing.java
package com.algolia;

import com.algolia.search.APIClient;
import com.algolia.search.ApacheAPIClientBuilder;
import com.algolia.search.Index;
import com.algolia.search.exceptions.AlgoliaException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.net.URL;
import java.util.List;

public class Indexing {

  public static void main(String[] args) throws IOException, AlgoliaException {
    List<Product> products =
      new ObjectMapper()
        .readValue(new URL("https://alg.li/doc-ecommerce.json"), new TypeReference<List<Product>>() {});

    APIClient client =
      new ApacheAPIClientBuilder("YourApplicationID", "YourAdminAPIKey").build();

    Index<Product> index = client.initIndex("demo_ecommerce", Product.class);
    index.addObjects(products);
  }

}
